web/client/src/routes/+page.svelte

```diff
@@ -0,0 +1,374 @@
+    event.preventDefault();
+  }
+
+  function detectRegion() {
+    if (browser) {
+      // attempt to detect region automatically
+      const browserLanguage = window?.navigator?.language;
+      if (browserLanguage && browserLanguage.split('-').length > 1) {
+        const code = navigator.language.split('-')[1];
+        region = code;
+      }
+    }
+  }
+
+  /** @param {DragEvent} event */
+  function handleDrop(event) {
+    event.preventDefault();
+    clearFile();
+
+    const files = event.dataTransfer?.files;
+    if (files) {
+      if (files.length < 1) {
+        addError('Sorry, you can only drop a ZIP file here.');
+      } else if (files.length > 1) {
+        addError('Sorry, you must upload only one ZIP file.');
+      } else {
+        handleFile(files[0], true);
+      }
+    }
+  }
+
+  /** @param {Event} event */
+  function handleFileInput(event) {
+    // @ts-ignore - TODO fix the types here?
+    const files = event?.target?.files;
+    handleFiles(files);
+  }
+
+  /**
+   * @param {File} file
+   * @param {boolean} shouldUpload
+   **/
+  function handleFile(file, shouldUpload = false) {
+    console.log('handling file', file.name);
+    clearFile();
+
+    if (file.type != 'application/zip') {
+      console.log('file type error', file.type);
+      addError('Sorry, only ZIP files are supported at this time.');
+      console.log(errors);
+    } else {
+      pendingFilename = file?.name;
+      if (shouldUpload) {
+        uploadFile(file);
+      }
+    }
+  }
+
+  /** @param {FileList} files */
+  function handleFiles(files, shouldUpload = false) {
+    console.log('handling files', files);
+    clearFile();
+    if (files.length > 1) {
+      addError('Sorry, you must upload only a single ZIP file.');
+    } else if (files.length == 1) {
+      handleFile(files[0], shouldUpload);
+    }
+  }
+
+  /** @param {Event} event */
+  function handleReset(event) {
+    event.preventDefault();
+    console.log('handleReset. region is', region);
+    // store selected region since it's unlikely to change
+    const selectedRegion = region;
+    if (form) form.reset();
+
+    // restore values after native reset
+    setTimeout(() => {
+      console.log('timeout', region);
+      region = selectedRegion;
+      sourceUrl = '';
+      clearFile();
+    }, 0);
+  }
+
+  /** @param {SubmitEvent} event */
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const file = fileInput?.files?.item(0);
+
+    if (file) {
+      handleFile(file, true);
+    } else if (sourceUrl) {
+      handleUrl(sourceUrl)
+    } else {
+      addError('Please include a file to validate.');
+    }
+  }
+
+  /** @param {string=} url **/
+  function createJob(url) {
+    return new Promise((resolve, reject) => {
+      /** @type {CreateJobParameters} */
+      const data = {
+        countryCode: region
+      };
+
+      if (url) {
+        data.url = url
+      }
+
+      const xhr = new XMLHttpRequest();
```

Comment:
Are you targeting a specific browser version set?  Wondering if you can get away with `fetch` here.

Reply:
We ran into some issues with fetch during testing. The upload request to GCS has to be configured in a very specific way to be accepted. I'd prefer to leave this as is.